### PR TITLE
Feature/gritas proc

### PR DESCRIPTION
**Pull request by CEgerer93**
Initial commits supporting processing of ods files through gritas. Markdown included, though it may need to be migrated elsewhere in time.

### CODE REVIEW

File: src/Applications/GrITAS_Proc/README.md [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+
+| Arguments        | Explanation   | Notes  |
+| ---------------- |:-------------:| -----:|
+| YR INI   | year of initial observations | Specify as standard four-digit year (ex. 2015) |
+| MNTH INI | month of initial observations| Specify as either one- or two-digit month (ex. 3 or 03 for March) |
+| DAY INI  | day of initial observations  | Specify as either one- or two-digit day (ex. 5 or 05 for 5th day of month)|
+| YR FIN   | year of final observations   | [See above] |
+| MNTH FIN | month of final observations  | [See above] |
+| DAY FIN  | day of final observations    | [See above] |
+| OBS SYS  | Observing system             | Currently supported: GEOSIT, GEOSFP, MERRA2
```
Comment (rtodling) [inline src/Applications/GrITAS_Proc/README.md]: Colin, what you are calling here OBS SYS is really what we call EXPID - that is the experiment identifier.

File: src/Applications/GrITAS_Proc/README.md [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+
+Upon error-free termination, four space-delimited `list` files will be generated, one for each synoptic time, cataloging all instrument `*ods` files available within the specified date range. Each list is of the form:
+```
+<OBS SYS>.<YYYYMMDD INI>-<YYYYMMDD FIN>.H<SYNOPTIC TIME>.list
```
Comment (rtodling) [inline src/Applications/GrITAS_Proc/README.md]: Same here: OBS SYS should be EXPID
Changing here truly means changing the actual bash and corresponding py script; minor change

File: src/Applications/GrITAS_Proc/README.md [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+```
+
+| Arguments        | Explanation   | Notes  |
+| ---------------- |:-------------:| -----:|
+| CONFIDENCE       | Confidence level of measurements (*need to check this one*) | Defaults to `0.95` |
+| DATE             | Forward slash delimited starting & ending dates | Format `YYYYMMDD/YYYYMMDD`, with `[INI DATE]`/`[FIN DATE]` |
+| PREFEXPDIR       | Directory prefix wherein observations (ods files) are located | User must currently turn to `obsSysLoc()` function of `driver.sh` for precise locations associated with support observing systems.|
+| PREFGRITAS       | Location of GrITAS source code   | User specified |
+| SYSOBSLISTPREFIX | Prefix of list files produced by `driver.sh`  | Example: `GEOSIT.20170101-20170103` for the GEOSIT observing system between 2017-01-01 & 2017-01-03 |
+| SYSTEM           | Observing system    | Currently supported: GEOSIT, GEOSFP, MERRA2 |
```
Comment (rtodling) [inline src/Applications/GrITAS_Proc/README.md]: SYSTEM should again be EXPID

File: src/Applications/GrITAS_Proc/README.md [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+###### To-Do: Improvements
+- [ ] Set `PREFEXPDIR` internally through an external call to `obsSysLoc()` function within `driver.sh`.
+- [ ] Set `PREFGRITAS` as a global export statement when environment is sourced.
+- [ ] Report to user combinations of `*ods` and `*rc` files that do not have matches.
+- [ ] Allow user to amend `convObs` list
+
+<br>
+<br>
+
+### ```grit.py```
```
Comment (rtodling) [inline src/Applications/GrITAS_Proc/README.md]: I think this program should eventually be enabled to also handle grass.x - we haven't talked about this, but this is the code the simply grids the data and writes out the observations to a corresponding nc4 file - that is, grisas.x does not to statistics ... and it does is gridding - statistics can in principle be done after the fact - this is useful for validation tests; system verification; all kinds of other things. 

So my suggestion here is to rename grit.py to grdstats.py- for gridded statistics - btw: 

gritas stands for Gridding Innovation Time Average Statistics
and
grisas stands for Gridding Innovation Synoptic Averaged Statistics

the average in grisas is simply a box average to put the observations in a grid, that is, ave all ob falling on the same grid and pretend they are one on that grid.

The fact is that gritas/sas have expand to be way more than simply about innovations - innovations = observation-minus-background differences ...

**Comment by rtodling** [comment]
After you make the minor changes I suggest, I will take it in and we can talk about the to-do. This is very nice.

**Review by rtodling — DISMISSED** [review]
Let's go from here.

**Comment by rtodling** [comment]
Colin: one more thing. earlier today I committed an update to the components.yaml file sitting on the top directory. You might want to take that update - it's in the branch you created.

**Review by CEgerer93 — COMMENTED** [review]
Noted the swap from 'tag' to 'branch' for building.

File: src/Applications/GrITAS_Proc/build-yaml.py [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+parser.add_option("-c","--confidence",type=float,default=0.95,
+                  help='Measurements confidence level (default = 0.95)')
+parser.add_option("-d","--date",type=str,default='YYYYMMDD/YYYYMMDD',
+                  help='Start/End dates of measurements - delimited via "/" (default = YYYYMMDD/YYYYMMDD)')
+parser.add_option("-e","--prefExpDir",type=str,default='./',
+                  help='Prefix of directories containing measurements (default = "./")')
+parser.add_option("-g","--prefGrITAS",type=str,default='./',
+                  help='Prefix of GrITAS src (default = ./)')
+parser.add_option("-l","--expObsListPrefix",type=str,default='foo.list',
```
Comment (CEgerer93) [inline src/Applications/GrITAS_Proc/build-yaml.py:19]: Changed sysObsListPrefix to expObsListPrefix to reflect suggestion of replacing 'OBS SYS' and 'SYSTEM' with 'EXPID'.

File: src/Applications/GrITAS_Proc/build-yaml.py [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+
+conf=options.confidence
+pandasDates=pd.date_range(start=options.date.split('/')[0],end=options.date.split('/')[1],freq='D')
+startDate=pandasDates[0].strftime('%Y-%m-%d')
+endDate=pandasDates[-1].strftime('%Y-%m-%d')
+system=options.system.upper()
+
+
+# List of instruments/dates/expIDs'
+# Determined by opening each expObsList file corresponding to desired synoptic time,
```
Comment (CEgerer93) [inline src/Applications/GrITAS_Proc/build-yaml.py:39]: Changed sysObsListPrefix to expObsListPrefix to reflect suggestion of replacing 'OBS SYS' and 'SYSTEM' with 'EXPID'.

File: src/Applications/GrITAS_Proc/build-yaml.py [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+# ..and successively using pandas.merge (def. inner) to determine intersection of all available instrument data
+instrDatesIDs=None
+for n,t in enumerate(options.synopticTimes.split('/')):
+    with open(options.expObsListPrefix+'.H%s.list'%t,'r') as f:
```
Comment (CEgerer93) [inline src/Applications/GrITAS_Proc/build-yaml.py:43]: Changed sysObsListPrefix to expObsListPrefix to reflect suggestion of replacing 'OBS SYS' and 'SYSTEM' with 'EXPID'.

File: src/Applications/GrITAS_Proc/README.md (hunk 1 above)
Comment (CEgerer93) [reply src/Applications/GrITAS_Proc/README.md]: Changed OBS SYS for EXPID, and 'Observing system' for 'experiment identifier'.

File: src/Applications/GrITAS_Proc/README.md (hunk 2 above)
Comment (CEgerer93) [reply src/Applications/GrITAS_Proc/README.md]: Changed OBS SYS for EXPID. Python and bash scripts changed separately, where appropriate.

File: src/Applications/GrITAS_Proc/README.md (hunk 3 above)
Comment (CEgerer93) [reply src/Applications/GrITAS_Proc/README.md]: Changed SYSTEM for EXPID, and 'Observing system' for 'experiment identifier'.

File: src/Applications/GrITAS_Proc/README.md (hunk 4 above)
Comment (CEgerer93) [reply src/Applications/GrITAS_Proc/README.md]: Indeed, I recall our conversations about including grisas.x functionality herein as a close counterpart to gritas.x. I have renamed grit.py to grdstats.py, as suggested.

File: src/Applications/GrITAS_Proc/driver.sh [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+    echo "Exp ID     = $1"
+    echo "System     = $2"
+    echo "Instrument = $3"
+    echo "DATE       = $4"
+    exit 2
+}
+
+
+if [ $# -ne 7 ]; then
+    echo "Usage: $0 <YR INI> <MNTH INI> <DAY INI> <YR FIN> <MNTH FIN> <DAY FIN> <OBS SYS>"
```
Comment (CEgerer93) [inline src/Applications/GrITAS_Proc/driver.sh]: Changed OBS SYS in favor of EXPID.

File: src/Applications/GrITAS_Proc/driver.sh [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+    echo `obsSysLoc`
```
Comment (CEgerer93) [inline src/Applications/GrITAS_Proc/driver.sh]: obsSysLoc replaced with expIDLoc.

File: src/Applications/GrITAS_Proc/driver.sh [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+    if [ $# -eq 0 ]; then
+	echo "Accepted observing systems: GEOSIT, GEOSFP, MERRA2"
+    elif [ $1 == 'GEOSIT' ]; then echo "/home/dao_ops/d5294_geosit_jan18/run/.../archive/obs"
+    elif [ $1 == 'GEOSFP' ]; then echo "/home/dao_ops/f516_fp/run/.../archive/obs"
+    elif [ $1 == 'MERRA2' ]; then echo "/home/dao_ops/d5124_m2_jan10/run/.../archive/obs"
+    else
+	echo "Unsupported observing system"
+	exit 2
+    fi
+}
```
Comment (CEgerer93) [inline src/Applications/GrITAS_Proc/driver.sh]: Function obsSysLoc replaced with expIDLoc; surrounding description also changed to reflect experiment IDs.

File: src/Applications/GrITAS_Proc/driver.sh [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+
+# Parse CL Args
+#--------------------------
+YR_I=$1; MNTH_I=$2; DAY_I=$3
+YR_F=$4; MNTH_F=$5; DAY_F=$6
+
+
+# Location of observations
+#--------------------------
+BASE=$(obsSysLoc $7); STATUS=$?
```
Comment (CEgerer93) [inline src/Applications/GrITAS_Proc/driver.sh]: obsSysLoc replaced with expIDLoc.

File: src/Applications/GrITAS_Proc/driver.sh [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+
+# Function to set list prefix
+#------------------------------
+getList()
+{
+    if   [ $1 == 'geosit' ]; then echo GEOSIT
+    elif [ $1 == 'geosfp' ]; then echo GEOSFP
+    elif [ $1 == 'merra2' ]; then echo MERRA2
+    else
+	echo "Unmatched instrument = $1"
```
Comment (CEgerer93) [inline src/Applications/GrITAS_Proc/driver.sh]: Instrument should be experiment identifier, again.

**Review by rtodling — APPROVED** [review]
Thanks for working on these.

**Pull request merged**